File: apollo/src/Request.ts
import mongoose from 'mongoose';
import { RequestType, RequestStatus } from './enums.js';

const Schema = mongoose.Schema;


const requestSchema = new mongoose.Schema({
    type: Object.values(RequestType), 
    sender: { type: Schema.Types.ObjectId, ref: 'User' },
    receivers: [{ type: Schema.Types.ObjectId, ref: 'User' }],
    acceptedBy: { type: Schema.Types.ObjectId, ref: 'User' },
    status: Object.values(RequestStatus),
    requestAttempts: BigInt,
    fromOrganizer: Boolean,
    event: { type: Schema.Types.ObjectId, ref: 'Event' },
    team: { type: Schema.Types.ObjectId, ref: 'Team' },    
    forRole: String,
    sentAt: String,
    acceptedAt: String,
});

const Request = mongoose.model('Request', requestSchema);

export default Request;

File: lib/models/appModels/Request.dart
import '../enums/RequestStatus.dart';
import '../enums/RequestType.dart';
import 'User.dart';
import 'Event.dart';
import 'Team.dart';

class Request {
  RequestType? type;
  User? sender;
  List<User>? receivers;
  User? acceptedBy;
  RequestStatus? status;
  int? requestAttempts;
  bool? fromOrganizer;
  Event? event;
  Team? team;
  String? forRole;
  int? sentAt;
  int? acceptedAt;

  Request({
    this.type,
    this.sender,
    this.receivers,
    this.acceptedBy,
    this.status,
    this.requestAttempts,
    this.fromOrganizer,
    this.event,
    this.team,
    this.forRole,
    this.sentAt,
    this.acceptedAt,
  });
}